### DISCUSSION TITLE
using ocilib with Visual Studio 6 C++

### DISCUSSION THREAD

**Discussion by vassilux** [vrogier/ocilib#296] [Ideas]
Hi,
I want to use ocilib in an old Visual Studio 6 project to migrate from ODBC to OCI, the C part it compiles and runs fine. Is it a good idea to mix runtime?

**Comment by vrogier** [comment]
Hi,

Mixing runtimes can be an issue when you're linking against them dynamically and/or when a memory block is allocated by one runtime and deallocated in another runtime. 

- OCILIB windows binaries are linked statically against msvc runtime. 
- There is no memory allocated in OCILIB and freed by application or vice versa.

Thus, it is safe :)

Best regards,

Vincent

**Reply by vassilux** [reply]
Hi Vincent,
thank for your answer
